app/services/enqueue_index_jobs.rb

```diff
@@ -0,0 +1,94 @@
+# Indexes a book version thru the SQS queuing system
+class EnqueueIndexJobs
+  def initialize
+    @todo_jobs_queue = TodoJobsQueue.new
+    @new_create_index_jobs = 0
+    @new_delete_index_jobs = 0
+    @worker_asg = AutoScalingGroup.new(worker_asg_name)
+  end
+
+  def call
+    released_book_ids.each do |book_id|
+      ACTIVE_INDEXING_VERSIONS.each do |indexing_version|
+        existing_book_indexing = find_book_indexing(book_id, indexing_version)
+
+        if existing_book_indexing
+          existing_book_indexing.in_demand = true
+        else
+          enqueue_create_index_job(book_id, indexing_version)
+        end
+      end
+    end
+
+    _, unneeded_book_indexings = book_indexings.partition(&:in_demand)
+
+    unneeded_book_indexings.each do |unneeded_book_indexing|
+      enqueue_delete_index_job(unneeded_book_indexing)
+    end
+
+    @worker_asg.increase_desired_capacity(by: new_jobs)
+
+    stats
+  end
+
+  private
+
+  def stats
+    {
+      num_delete_index_jobs: @new_delete_index_jobs,
+      num_create_index_jobs: @new_create_index_jobs
+    }
+  end
+
+  def book_indexings
+    @book_indexings ||= BookIndexState.live_book_indexings
+  end
```

Comment:
Change name of this method/variable and other uses in this class away from "BookIndexing" to match name change to BookIndexState, e.g. `def index_states`